### PR TITLE
Task/hw 53101 core list transfers

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

File: Sources/Model/Transfer/HyperwalletTransferQueryParam.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+/// Representstion of the user transfers QueryParams fields.
```
Comment (jsaini-hw) [inline Sources/Model/Transfer/HyperwalletTransferQueryParam.swift]: typo in Representstion

File: Tests/Responses/Transfer/ListTransferResponse.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+    "count": 23,
```
Comment (jsaini-hw) [inline Tests/Responses/Transfer/ListTransferResponse.json]: data in the response is 2 so count should be 2 too

File: Tests/Responses/Transfer/ListTransferResponse.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    ],
+    "links": [
+        {
+            "params": {
+                "rel": "self"
+            },
+            "href": "https://localhost/rest/v3/transfers/?offset=0&limit=10&sourceToken=usr-123456"
+        },
+        {
+            "params": {
```
Comment (jsaini-hw) [inline Tests/Responses/Transfer/ListTransferResponse.json]: There's no next and last params if total data is less than limit and offset is 0

File: Tests/Transfer/HyperwalletTransferTests.swift [hunk 4]
```diff
@@ -138,6 +138,30 @@ class HyperwalletTransferTests: XCTestCase {
         XCTAssertNotNil(statusTransitionResponse, "The `statusTransitionResponse` should not be nil")
         verifyStatusTransitionResponse(statusTransitionResponse)
     }
+
+    func testListTransfer_success() {
```
Comment (jsaini-hw) [inline Tests/Transfer/HyperwalletTransferTests.swift]: update name to testListTransfers
add testListTransfers_emptyResult similar to testListBankAccounts_emptyResult

File: Sources/Model/Transfer/HyperwalletTransferQueryParam.swift (hunk 1 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransferQueryParam.swift]: Fixed

File: Tests/Responses/Transfer/ListTransferResponse.json (hunk 2 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Responses/Transfer/ListTransferResponse.json]: Fixed

File: Tests/Responses/Transfer/ListTransferResponse.json (hunk 3 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Responses/Transfer/ListTransferResponse.json]: Removed

File: Tests/Transfer/HyperwalletTransferTests.swift (hunk 4 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransferTests.swift]: Renamed. Added `testListTransfers_emptyResult` test

**Pull request merged**